File: python/2/18.py
import sys
import pandas as pd

filename = "../../data/hightemp.txt"
df = pd.read_csv(filename, header=None, delimiter='\t',
        names=('pref', 'city', 'temp', 'date'))
df = df.sort_values(by=['temp', 'city'], ascending=False)
df.to_csv(sys.stdout, index=False, header=False, sep='\t')

File: ruby/2/18.rb
require "csv"

filename = "../../data/hightemp.txt"
table = CSV.table(filename, col_sep: "\t", headers: %w(pref city temp day))

sorted = table.sort_by { |row| -row[:temp] }
sorted.each do |row|
  puts row.to_s.gsub(/,/, "\t")
end
